apps/frontend/core-dapp/src/features/trade/TradeStore.ts

```diff
@@ -347,6 +354,19 @@ export class TradeStore {
     )
   }
 
+  get closePositionNeedPermit(): boolean | undefined {
+    if (!TRADE_VIA_HELPER_ENABLED) return false
+
+    if (!this.selectedPosition || !this.root.web3Store.connected) return false
+    if (this.closePositionAmount === undefined) return undefined
+    if (this.selectedMarket?.resolved) return false
```

Comment:
Would this change affect UX? If not we should group them for readability, lmkwyt
```suggestion
    if (!this.selectedPosition || !this.root.web3Store.connected || this.selectedMarket?.resolved) return false
    if (this.closePositionAmount === undefined) return undefined
```

Reply:
Good catch, I don't think it does affect anything